src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splitElement = fromFormat[3];
+  const joinElement = toFormat[3];
+  const newDate = date.split(splitElement);
+
+  let formatAgan = [];
+
+  let year = formatAgan[0];
+  let month = formatAgan[1];
+  let day = formatAgan[2];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    if (fromFormat[i] === 'YY' || fromFormat[i] === 'YYYY') {
+      year = newDate[i];
+    }
+
+    if (fromFormat[i] === 'YYYY') {
+      year = newDate[i];
+    }
+
+    if (fromFormat[i] === 'MM') {
+      month = newDate[i];
+    }
+
+    if (fromFormat[i] === 'DD') {
+      day = newDate[i];
+    }
+  }
+
+  if (toFormat.includes('YYYY') && year.length === 2 && year < 30) {
+    year = `${20}` + `${year}`;
+  }
+
+  if (toFormat.includes('YYYY') && year.length === 2 && year >= 30) {
+    year = `${19}` + `${year}`;
+  }
+
+  if (toFormat.includes('YY')) {
+    year = year.toString().slice(-2);
+  }
+
+  for (let i = 0; i < toFormat.length; i++) {
```

Comment:
The loop iterating over `toFormat` starts at index 0 which is correct, but it's better to use `toFormat.length - 1` as the condition to iterate, similar to how you did with `fromFormat`. This would avoid including the separator in the iteration, which isn't necessary.